src/hooks/use-auth.ts

```diff
@@ -0,0 +1,13 @@
+import { useSelector } from 'react-redux';
+
+export const useAuth = () => {
+  const { email, token, id, login } = useSelector(state => state.user);
```

Comment:
do not use [useSelector destructuring](https://stackoverflow.com/questions/59072200/useselector-destructuring-vs-multiple-calls), use useSelector only in the component where it is needed